lib/Doctrine/DBAL/Platforms/Keywords/MySQL8Keywords.php

```diff
@@ -0,0 +1,311 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Platforms\Keywords;
+
+/**
+ * MySQL 8 reserved keywords list.
+ *
+ * @author Michele Locati <michele@locati.it>
+ * @link   www.doctrine-project.org
+ */
+class MySQL8Keywords extends MySQL57Keywords
```

Comment:
To follow similar naming pattern, this class should be named MySQL80Keywords.
Also if possible, should also only add new keywords (or remove removed ones), like here: https://github.com/doctrine/dbal/blob/9f3188ab3d56db6f6fc76c2ddf6b0b4aae06192f/lib/Doctrine/DBAL/Platforms/Keywords/PostgreSQL94Keywords.php#L49-L55

Reply:
> To follow similar naming pattern, this class should be named MySQL80Keywords.

:+1:

> Also if possible, should also only add new keywords (or remove removed ones), like here:

What about the current `MySQL57Keywords` class? Should it be updated to have the same behavior vs its underlying `MySQLKeywords` class

Reply:
Furthermore: if a keywords will be added to `MySQL57Keywords`, and it's already present in `MySQL80Keywords`, with the [approach you suggest](https://github.com/doctrine/dbal/blob/9f3188ab3d56db6f6fc76c2ddf6b0b4aae06192f/lib/Doctrine/DBAL/Platforms/Keywords/PostgreSQL94Keywords.php#L49-L55) we'll have duplicated kewords: should be call `array_unique` (and `array_values` to keep the array indexes sane)?